nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/SimpleRecordSchema.java

```diff
@@ -167,6 +167,11 @@ public boolean equals(final Object obj) {
         }
 
         final RecordSchema other = (RecordSchema) obj;
+        if (getSchemaNamespace().isPresent() && getSchemaNamespace().equals(other.getSchemaNamespace()) &&
+                getSchemaName().isPresent() && getSchemaName().equals(other.getSchemaName())) {
+            return true;
+        }
+
```

Comment:
I think it would worth to extend `TestSimpleRecordSchema#testHashCodeAndEqualsWithSelfReferencingSchema` with this case.

Reply:
Thanks @simonbence for the comment, I added a new test case to the TestSimpleRecordSchema

Reply:
Thanks for taking care of it!